package.json

```diff
@@ -183,6 +183,7 @@
     "thunky": "^1.1.0",
     "typescript": "^3.7.5",
     "untildify": "^4.0.0",
+    "vasync": "^2.2.0",
```

Comment:
Yah, I understand our hesitance to use a new dependency. I figured this was okay because it is in devDeps, only.  Note that the agent doesn't currently have a general "handle async flow" dependency, but instead has more use-case-specific `after-all-results`, `async-value-promise`.  Granted `vasync` isn't really maintained anymore and is waaay less popular than `async`, I've found it more reliable than `async` --- but I'm totally biased given that `vasync` came from my last gig.